src/CanvasRenderingContext2d.cc

```diff
@@ -1802,8 +1802,17 @@ NAN_GETTER(Context2d::GetShadowColor) {
 
 NAN_METHOD(Context2d::SetFillColor) {
   short ok;
-  if (!info[0]->IsString()) return;
-  String::Utf8Value str(info[0]);
+
+  Isolate *isolate = Isolate::GetCurrent();
+  v8::TryCatch try_catch(isolate);
+  MaybeLocal<String> color = info[0]->ToString(Context::New(isolate));
+
+  if (try_catch.HasCaught()) {
+    try_catch.ReThrow();
+    return;
+  }
+
+  String::Utf8Value str(color.ToLocalChecked());
```

Comment:
I think it'd be easier to just do `this._setFillColor(String(val))` in the JS file and let it throw there if it fails. But if you want to keep this in C++, can you switch to `Nan::TryCatch` please?